java/src/main/java/com/adhoc/flight/utils/QueryUtils.java

```diff
@@ -0,0 +1,241 @@
+package com.adhoc.flight.utils;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+
+public final class QueryUtils {
+
+    private QueryUtils() {
+        // Prevent instantiation.
+    }
+
+    public static final class Writer {
+
+        private Writer() {
+            // Prevent instantiation.
+        }
+
+        /**
+         * Writes the {@link ByteArrayOutputStream} representation of the provided
+         * {@link VectorSchemaRoot} to the provided {@link File}.
+         * 
+         * @param vectorSchemaRoot the {@code VectorSchemaRoot} to be converted.
+         * @param file the file to write to.
+         * @throws IOException If an error occurs when trying to write to the file.
+         */
+        public static void writeToBinaryFile(VectorSchemaRoot vectorSchemaRoot, File file) throws IOException {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
final markers.

Reply:
`ByteArrayOutputStream` now removed. All data is directly saved to the file now.